src/org/skypro/skyshop/articles/SearchEngine.java

```diff
@@ -14,11 +15,12 @@ public void add(Searchable searchObject) {
         search.add(searchObject);
     }
 
-    public List<Searchable> search(String query) {
-        List<Searchable> result = new ArrayList<>();
+    public TreeMap<String, Searchable> search(String query) {
```

Comment:
не забываем про полиморфизм
```suggestion
    public Map<String, Searchable> search(String query) {
```